app/server-utils/responses.ts

```diff
@@ -7,3 +7,29 @@ export const jsonCORS = <T>(data: T) =>
       'Access-Control-Allow-Origin': ALLOW_ORIGINS,
     },
   })
+
+export const CORSOptions = (request: Request) => {
+  const origin = request.headers.get('origin') || ''
+  const isOriginAllowed = ALLOW_ORIGINS == '*' || ALLOW_ORIGINS.split(',').includes(origin)
+  const allowed = isOriginAllowed ? origin : ''
+
+  return json(
+    {},
+    {
+      headers: {
+        'Access-Control-Allow-Origin': allowed,
+        'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE',
+        'Access-Control-Allow-Headers': 'Content-Type, Authorization, allow-control-allow-origin',
+      },
+    }
+  )
+}
+
+type Handler = (args: LoaderArgs) => any
+export const wrapCORS = (fn: Handler) => async (args: LoaderArgs) => {
+  if (args.request.method == 'OPTIONS') {
+    return CORSOptions(args.request)
```

Comment:
is any of this special to cloudflare or remix? if yes, a link to relevant documentation would be nice ... if not, if all just standard HTTP server response stuff 👍

Reply:
no. This is to get the chat bot working, which uses remix. This is all bog standard http stuff